### PR TITLE
Task #223691 feat: Cohort creation hierarchy in master and resolved UT issues

### PR DESCRIPTION
**Pull request by rahulg1254**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,3 +1,3 @@
-NEXT_PUBLIC_BASE_URL=https://qa.prathamteacherapp.tekdinext.com/user/v1
+NEXT_PUBLIC_BASE_URL=https://backend.prathamdigital.org/user/v1
```
Comment (itsvick) [inline .env:1]: @rahulg1254 dont push env changes

File: src/pages/block.tsx [hunk 2]
```diff
@@ -120,17 +137,34 @@ const Block: React.FC = () => {
...
-        console.log("Fetched fieldId:", blocksFieldId); // Check if this logs correctly
+
+        setPaginationCount(response?.result?.totalCount || 0);
+        // console.log("Fetched fieldId:", blocksFieldId);
+
+        const totalCount = response?.result?.totalCount || 0;
+        const pageCount = Math.ceil(totalCount / limit);
+        setPageCount(pageCount);
+
+        if (totalCount >= 15) {
```
Comment (itsvick) [inline src/pages/block.tsx:161]: @rahulg1254 get this all numbers from the app.constant

File: src/pages/district.tsx [hunk 3]
```diff
@@ -130,27 +152,38 @@ const District: React.FC = () => {
...
-      const data = {
-        limit: pageLimit,
-        offset: pageOffset * pageLimit,
-        controllingfieldfk: selectedState === "ALL" ? undefined : selectedState,
-        fieldName: "districts",
+    await fetchDistrictData(selectedStateDropdown);
+
+    if (selectedStateDropdown) {
+      const reqParams = {
+        limit: 10,
```
Comment (itsvick) [inline src/pages/district.tsx:163]: @rahulg1254 here as well

File: src/pages/district.tsx [hunk 4]
```diff
@@ -130,27 +152,38 @@ const District: React.FC = () => {
...
+        offset: 0,
+        filters: {
+          name: "Maharashtra",
+          type: "STATE",
```
Comment (itsvick) [inline src/pages/district.tsx:167]: @rahulg1254 why this is hardcoded?

File: public/locales/en/common.json [hunk 5]
```diff
@@ -64,6 +64,11 @@
     "STATE_NAME_REQUIRED": "State Name is Required",
     "DISTRICT_NAME_REQUIRED": "District Name is Required",
     "CREATE":"Create",
+    "BLOCK_ADDED_SUCCESS": "Block Added Successfully!",
+    "BLOCK_ADDED_FAILURE": "Block is not Added!",
```
Comment (itsvick) [inline public/locales/en/common.json:68]: @rahulg1254 update this message too, Unable to add Block

File: public/locales/en/common.json [hunk 6]
```diff
@@ -64,6 +64,11 @@
...
+    "BLOCK_UPDATED_SUCCESS": "Block Updated Successfully!",
+    "BLOCK_UPDATED_FAILURE": "Block is not Updated!",
```
Comment (itsvick) [inline public/locales/en/common.json:70]: @rahulg1254 Unable to update block

File: .env (hunk 1 above)
Comment (rahulg1254) [reply .env:1]: resolved

File: src/pages/district.tsx (hunk 4 above)
Comment (rahulg1254) [reply src/pages/district.tsx:167]: Resolved

**Pull request merged**